qiskit/primitives/base/base_result.py

```diff
@@ -45,6 +45,12 @@ def __post_init__(self) -> None:
             TypeError: If one of the data fields is not a Sequence or ``numpy.ndarray``.
             ValueError: Inconsistent number of experiments across data fields.
         """
+        warn(
+            "The class ``BasePrimitiveResult`` is deprecated as of qiskit 0.46.0. "
+            "It will be removed no earlier than 3 months after the release date. "
+            "Use PrimitiveResult in PrimitiveV2 instead.",
```

Comment:
I think it would help to show the path of the new `PrimitiveResult` class. From #11227 I take that would be:

```suggestion
            "Use PrimitiveResult class in `qiskit.primitives.containers` instead.",
```